source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+// https://localhost:8080/users/profile/edit  <-- matches
+// https://localhost:8080/users/transaction?month=03&year=2020 <-- matches
+
+```
+
+However, if you need to match additional properties on the route, you can pass an object instead.
+
+```js
+cy.route2({
+  path: '/users/**',
+  method: 'POST',
+})
+```
+
+The following contains a complete list of available properties you can verify against:
+
+Option | Default | Description
+--- | --- | ---
+`auth` | `null` | Object with a `username` and `password` to match HTTP basic authentication
+`headers` | `null` | Object to match client request headers
+`hostname` | `null` | String, RegExp, or Glob to match based on requested hostname
+`method` | `'ALL'` | Match based on HTTP method (`GET`, `POST`, `PUT`, etc.)
+`path` | `null` | String, RegExp, or Glob to match on request path after the hostname, including query params
+`pathname` | `null` | String, RegExp, or Glob to match on request path after the hostname, without query params
+`port` | `null` | Match based on requested port number
+`query` | `null` | Object to match based on query parameters
+`url` | `null` | String, RegExp, or Glob to match based on full request URL
+
+**{% fa fa-angle-right %} response** ***(String, Object, Array, Function)***
+
+This allows you to supply a response `body` to {% url 'stub' stubs-spies-and-clocks#Stubs %} in the matching route. This is accomplished by giving you access to the request that you can use in a plain function.
+
+```js
+cy.route2('/users/**', (req) => {
+  req.send({
+    statusCode: 200,
+    body: JSON.stringify({
+      profile: {
+        firstName: 'Tony',
+        lastName: 'Jarvis'
+      }
+    })
+  })
+})
+```
+
+In addition, because you have access to the request, this means that you can modify the request as well:
+
+```js
+cy.route2('/users/**', (req) => {
+  req.headers = { ...req.headers, accept: 'application/json' }
+  req.body = { ...req.body, note: 'Custom note' }
+})
+```
+
+**{% fa fa-angle-right %} method** ***(String)***
+
+Match the route to a specific HTTP method (`GET`, `POST`, `PUT`, etc).
+
+{% note bolt %}
+By default, Cypress will match `ALL` requests by default.
+{% endnote %}
+
+## Yields {% helper_icon yields %}
+
+{% yields null_alias cy.route2 %}
+
+# Examples
+
+## Without Stubbing
+
+If you do not pass a `response` to a route, Cypress will pass the request through without stubbing it. We can still wait for the request to resolve later.
+
+### Wait on `GET` request matching `url`
+
+```javascript
+cy.route2('**/users').as('getUsers')
+cy.visit('/users')
+cy.wait('@getUsers')
+```
+
+### Wait on matching `method` and `url`
+
+```javascript
+cy.route2('POST', '**/users').as('postUser')
+cy.visit('/users')
+cy.get('#first-name').type('Julius{enter}')
+cy.wait('@postUser')
+```
+
+### Wait on `url` matching glob
+
+Under the hood Cypress uses {% url 'minimatch' https://github.com/isaacs/minimatch %} to match glob patterns of `url`.
```

Comment:
`minimatch` support applies to any `S` string property on `RouteMatcher` as well: https://github.com/cypress-io/cypress/blob/50a45c1b707ef0ccb0447b94ecb04946e994a4db/packages/net-stubbing/lib/external-types.ts#L125-L167 this includes strings in the `auth`, `headers`, and `query` objects

Reply:
as mentioned earlier, this section could also be updated to note that minimatch is used for any `StringMatcher` field